### ISSUE TITLE
I055 — Remove Future.detach and enforce strict structured child lifetime

### ISSUE THREAD

**Issue by guillermomolina** [guillermomolina/protos#656]
## Implementation owner

**Formal identifier:** `I055`  
**Parent:** #623  
**State:** `READY`  
**Decision dependency:** D159 / #623  
**Ratified decision record:** `guillermomolina/protos-project-docs:docs/project/decisions/language/D159_FUTURE_DETACHMENT_AND_STRICT_STRUCTURED_CHILD_LIFETIME.md`  
**Ratification revision:** `3a40a27b4784945e47ca6e4e9d735447a2326f8c`

## Purpose

Implement ratified D159 Candidate B without reopening the decision:

```text
Future.detach()                              REMOVE_NOW_RECONSIDER_LATER
task-backed child ownership                  STRICT_COMPLETE_LIFETIME
background/independent Actor-local work       DEFERRED
public Task/scope                             REMAINS ABSENT
```

D159 intentionally does not select a replacement background-work mechanism.

## Required semantic result

After I055:

- Core exposes no `Future.detach()` selector;
- a task-backed child created in the current asynchronous task scope remains owned by that scope until the child is terminal;
- returning, storing, wrapping, or passing the child Future does not change ownership;
- Future remains the public eventual-result/cancellation/observation handle;
- Actor lifetime remains a separate semantic boundary;
- no public Task or structured-scope object is introduced;
- `Future.then` retains downstream-only cancellation/ownership except for removal of obsolete detachment wording;
- non-task-backed and terminal Future detachment no-op semantics cease to exist because the operation itself is absent.

## Normative reconciliation

At minimum inspect and reconcile current-main references in:

- `spec/concurrency/FUTURES_AND_TASKS.md`;
- `spec/runtime/ABSTRACT_RUNTIME.md`;
- `spec/semantics/CALLABLES.md` where Future-producing execution ownership is referenced;
- `spec/PROTOS_SPEC_CHANGELOG.md`;
- `docs/guide/08-futures-and-structured-concurrency.md`;
- any README/guide text that exposes `detach()`.

Do not alter D045 task-scoped structured ownership except to make the child ownership edge strict for the child's complete lifetime.

## Runtime scope

At minimum inspect and reconcile:

- `ProtosStandardFutureProtocol` — remove the public selector;
- `ProtosFutureValue` — remove detached state and public detachment behavior;
- `ProtosTask.detachFromParent()` — remove it if no remaining runtime-internal consumer requires it;
- normal parent/child terminalization and child-drain behavior — preserve;
- Actor-incarnation termination behavior — preserve;
- Future cancellation/adoption/`then`/`all` behavior — preserve unless text/tests refer specifically to detach.

Do not use this implementation slice to design creation-time independent work.

## Test reconciliation

Repository evidence shows current guest `.detach()` occurrences are principally conformance/control fixtures used to manufacture cancellation/cleanup dependencies.

Rewrite those fixtures so they continue to test their intended cancellation, cleanup, ordering, or child-drain semantics without relying on removed public detachment.

Add focused evidence that:

1. a pending task-backed child cannot escape its creator task's structured lifetime;
2. normal owner completion waits for all owned task-backed children;
3. child terminalization releases the parent edge normally;
4. Future return/store/wrap/pass does not transfer ownership;
5. Actor termination semantics are unchanged;
6. non-task-backed Future behavior remains correct without a detach selector;
7. invoking missing `detach` follows ordinary missing-selector behavior;
8. cancellation and `ensure` cleanup remain green;
9. `Future.then` downstream-only cancellation remains green;
10. `Future.all` remains unchanged.

## Deferred boundary

I055 MUST NOT select:

- a replacement background-work selector or syntax;
- a daemon/service task model;
- creation-time independent Actor-local work;
- a public Task/scope;
- a new Actor lifetime/supervision model.

If implementation exposes a genuine need for one of those, stop at the substantive design boundary and route a separate Dxxx.

## Validation

Follow current root/scoped `AGENTS.md`.

At minimum:

- focused Future/task/structured-ownership tests;
- focused cancellation and cleanup tests affected by fixture rewrites;
- full required repository validation;
- `git diff --check`;
- source-license compliance;
- implementation version/changelog rules for executable-source changes;
- specification changelog rules for normative changes;
- required CI green before closure under current repository policy.

## Publication boundary

Repository-content changes in `guillermomolina/protos` follow the interactive local contribution workflow. Do not bypass it with connector/API repository-content writes.

No force-push.

## Closure

I055 closes only when the D159 semantic reconciliation is published to `main`, all obsolete detachment surface is removed consistently, retained structured-concurrency behavior is green, and required validation/CI closure gates pass.

**Comment by guillermomolina** [comment]
## PUBLICATION / CLOSURE REVIEW

I055 product publication is present at `3fb863b44f1bf5b259abbb8fd44a45635a72f8fd` and the required durable record is published at `guillermomolina/protos-project-docs@dfcea737c68dfdc6c099b39c45da0791c5a9c61b`.

Durable reconciliation confirms:
- D159 Candidate B implemented;
- B008 is CLOSED;
- D045 task-scoped ownership remains the baseline and is preserved;
- the pre-existing `Future.then()` source-wait cancellation defect exposed by I055 is recorded as an I055 same-slice repair, not a separate BUG item.

Closure is intentionally withheld because inspection of the published product commit found two paths that the implementation handoff explicitly identified as unrelated to I055 but which are nevertheless included in that commit:
- `.devcontainer/Dockerfile`
- `.gitignore`

Repository publication safety requires the staged/publication set to be exactly the patch-owned paths. This mismatch must be reconciled explicitly before I055 is closed; no cleanup commit is being manufactured automatically.

`PROTOS_REVISION=3fb863b44f1bf5b259abbb8fd44a45635a72f8fd`
`PROJECT_RECORD_REVISION=dfcea737c68dfdc6c099b39c45da0791c5a9c61b`
`CROSS_REFERENCES=PASS`
`B008=CLOSED`
`D045_TASK_SCOPED_OWNERSHIP=PRESERVED`
`DURABLE_RECORD_DECISION=REQUIRED`
`REQUIRED_DURABLE_PUBLICATION=PASS`
`ISSUE_CLOSURE_COMMENT=PENDING_PUBLICATION_SCOPE_RECONCILIATION`

**Comment by guillermomolina** [comment]
## CLOSURE

Maintainer reconciliation: the unrelated `.devcontainer/Dockerfile` and
`.gitignore` paths in product revision
`3fb863b44f1bf5b259abbb8fd44a45635a72f8fd` were accidentally staged by the
maintainer through the IDE. The maintainer has explicitly accepted that
publication scope; it is not an I055 semantic or implementation defect and no
cleanup publication is required for I055 closure.

Final durable coordinates:

```text
PROTOS_REVISION=3fb863b44f1bf5b259abbb8fd44a45635a72f8fd
PROJECT_RECORD_REVISION=dfcea737c68dfdc6c099b39c45da0791c5a9c61b
CROSS_REFERENCES=PASS
B008=CLOSED
D045_TASK_SCOPED_OWNERSHIP=PRESERVED
DURABLE_RECORD_DECISION=REQUIRED
REQUIRED_DURABLE_PUBLICATION=PASS
CLOSURE_EVIDENCE_IDENTIFIED=PASS
ISSUE_CLOSURE_COMMENT=PASS
```

I055 is complete under the maintainer's explicit publication-scope reconciliation.
